packages/x-charts/src/internals/plugins/featurePlugins/useChartCartesianAxis/computeAxisValue.ts

```diff
@@ -249,3 +259,32 @@ export function computeAxisValue<T extends ChartSeriesType>({
     axisIds: allAxis.map(({ id }) => id),
   };
 }
+
+export function scaleRangeAndTickNumber(
+  scale: D3ContinuousScale,
+  minData: number,
+  maxData: number,
+  unscaledTickNumber: number,
+) {
+  const domain = scale.domain();
+  const scaleRange = scale.range();
+  const rangeSpan = Math.abs(scaleRange[1] - scaleRange[0]);
+  const domainSpan = Math.abs(domain[1].valueOf() - domain[0].valueOf());
+  const extremaSpan = Math.abs(maxData - minData);
+  const spanRatio = extremaSpan === 0 ? 1 : domainSpan / extremaSpan;
+  const startDiff = Math.abs(domain[0].valueOf() - minData);
+  const endDiff = Math.abs(domain[1].valueOf() - maxData);
+  const startRatio = startDiff / (startDiff + endDiff) || 0;
+  const endRatio = endDiff / (startDiff + endDiff) || 0;
+
+  const sign = scaleRange[1] <= scaleRange[0] ? 1 : -1;
+
+  const range = [
+    scaleRange[0].valueOf() + sign * rangeSpan * startRatio * (spanRatio - 1),
+    scaleRange[1].valueOf() - sign * rangeSpan * endRatio * (spanRatio - 1),
```

Comment:
Found a bug due to floating point arithmetic. 

https://github.com/user-attachments/assets/2d4932b1-32ee-44cd-9e44-105c21df1a98

The point at x:3, y: 4 isn't showing up because the `scale(4)` is `255.00000000000003`, which is outside the drawing area, which is `[255, 20]`.